### PR TITLE
Add --quiet option to upload_file.

### PR DESCRIPTION
**Pull request by bwbeach**
This allows scripts to easily parse the JSON that upload_file prints to stdout.

### CODE REVIEW

File: b2 [hunk 1]
```diff
@@ -801,9 +806,9 @@ def upload_file(args):
             headers['X-Bz-Info-' + k] = b2_url_encode(v)
 
         try:
-            response = post_file(url, headers, local_file, exit_on_error=False, progress_bar=True)
+            response = post_file(url, headers, local_file, exit_on_error=False, progress_bar=(not quiet))
```
Comment (ppolewicz) [inline b2]: The extra parenthesis look weird. I would make a new boolean variable `progress_bar` before the `try:` statement to make it easier to read. It doesn't matter very much, we'll clean it up next week anyway.

File: b2 [hunk 2]
```diff
@@ -801,9 +806,9 @@ def upload_file(args):
...
             print json.dumps(response, indent=4, sort_keys=True)
-            if 'fileId' in response:
+            if (not quiet) and ('fileId' in response):
```
Comment (ppolewicz) [inline b2]:
```python
if 'fileId' in response and not quiet:
```
has much less parenthesis and is much more readable (reads like english)

**Comment by ppolewicz** [comment]
The commit should have "Closes #23" or "Resolves #23" in the message so that #23 is closed automatically upon merging.

**Comment by bwbeach** [comment]
I'll do the "Closes #23" next time.

File: b2 (hunk 2 above)
Comment (bwbeach) [inline b2]: As you can tell, I tend to prefer more parens when there might be any question about operator precedence.  This case could go either way.

Comment (ppolewicz) [inline b2]: True, but I find the suggested way to be more pythonic (and more readable to python coders) than the one you used, that is why I suggested it.

**Comment by ppolewicz** [comment]
In such case the last commit message can be simply changed by running `git commit --amend` and pushing the topic branch with `-f` flag, so `git push -f origin quiet_uploads`. (please don't push the master branch with `-f` or forks may break for other people).

**Pull request merged**